pkg/util/net.go

```diff
@@ -266,3 +274,55 @@ func GetHostIPv6(ipHint string) (string, error) {
 
 	return "", fmt.Errorf("unable to find host IPv6 address")
 }
+
+// Find the Default route Interface based on ipv4 or ipv6 routes.
+func FindDefaultRouteIface() (iface *tcpnet.Interface, err error) {
```

Comment:
I think we need to pass the configuration here to determine if MicroShift is running on ipv4, ipv6, or both.
Imagine you have different NICs in the host for each IP family. Default routes would be configured on different devices, therefore they could have different MTUs.
If MicroShift is running on ipv6 check only ipv6 routes.
If MicroShift is running on ipv4 check only ipv4 routes.
If MicroShift is running on both, check both and pick the smallest.
Or is there something I am missing?

Reply:
I'm not sure its a common scenario for edge devices when you have different Default routes set across different IP families .

Reply:
we are calling FindDefaultRouteIface from the ovn package which doesn't have its own configuration. the IPv4/6 detection logic is in the config [struct](https://github.com/praveenkumar/microshift/blob/offline_images/etcd/vendor/github.com/openshift/microshift/pkg/config/config.go#L582) .